### ISSUE TITLE
El Capitan Update Error

### ISSUE THREAD

**Issue by vinkla** [Homebrew/homebrew#41690]
Maybe this is to early to submit El Capitan errors but here we go.

When running `brew update` and the `brew upgrade --all` it returns the following error:

```bash
Error: Failure while executing: git pull --quiet origin refs/heads/master:refs/remotes/origin/master
```

```bash
Warning: You have uncommitted modifications to Homebrew
If this a surprise to you, then you should stash these modifications.
Stashing returns Homebrew to a pristine state but can be undone
should you later need to do so for some reason.
    cd /usr/local/Library && git stash && git clean -d -
```

Which is strange since I haven't touched those files. When running a hard reset I've no permission reset the repository files:

```bash
error: unable to unlink old '.gitignore' (Permission denied)
fatal: Could not reset index file to revision 'origin/master'.
```

Doing a reset to HEAD works fine but it doesn't get me any updates.

This is all very strange since it worked just fine yesterday. I'm using El Capitan public beta 15A215h and Homebrew 0.9.5. This is similar issues as in #41150 and #41665.

**Comment by vinkla** [comment]
Solved it by running `sudo chown $(USER):admin /usr/local`.

**Comment by ilkkao** [comment]
Noticed the same problem and fixed it manually as well.

**Comment by mcb** [comment]
This problem also occurs on 10.10.4 and Homebrew 0.9.5. Are you sure this is something El Capitan related?

**Comment by mcb** [comment]
Seems like this is more of an OSX-upgrade issue than El Capitan specific. Running the command above but with the ```-R``` option solved the issue for me as well.

**Comment by vinkla** [comment]
Maybe not El Capitan specific but I first experienced it on El Capitan. Had no problem when updating to 10.10.4 though. That this has to do with the OS X upgrade makes sense.

**Comment by DomT4** [comment]
Not seeing this personally on either 10.10 or 10.11.

If the workaround is `sudo chown -R $(USER):admin /usr/local` that's fairly pleasant at least. Closing as something beyond Homebrew's control, but will shunt any others reporting this problem to this Issue.

**Comment by phynet** [comment]
I'm having the same problem here with 10.11.2 Mac OS X El Capitán.

**Comment by DomT4** [comment]
```
sudo chown -R $(USER):admin /usr/local
```
Remains the advice whilst we continue work on an internal command so people don't have to issue a broad chown.

**Comment by phynet** [comment]
Ups, thank you, with the command `brew update` I saw the command to change the permission
to write in `/usr/local` and it seems it's working, sorry for the comment above.

**Comment by DomT4** [comment]
No worries! If anyone wants to track the likely long-term solution the PR is at https://github.com/Homebrew/homebrew/pull/45934.

**Comment by kitekite** [comment]
cd $(brew --prefix) && git fetch && git reset --hard origin/master. This fixed my brew update issue!

**Comment by jamieorc** [comment]
> cd $(brew --prefix) && git fetch && git reset --hard origin/master

This also solved same problem for me (OS X 10.11.3)

**Comment by gautamkishore** [comment]
> cd $(brew --prefix) && git fetch && git reset --hard origin/master

This solved the issue for me too. Thanks Kavitha @kitekite
